python/pyarrow/tests/test_pandas.py

```diff
@@ -1438,6 +1438,13 @@ def test_timestamp_to_pandas_ns(self):
     def test_timestamp_to_pandas_out_of_bounds(self):
         # ARROW-7758 check for out of bounds timestamps for non-ns timestamps
 
+        if Version("2.0.0") < Version(pd.__version__):
```

Comment:
```suggestion
        if Version(pd.__version__) >= Version("2.1.0.dev"):
```

I think this is more robust for when pandas 2.0.1 gets released (that would pass this check, while it doesn't need to be skipped)